private/buf/bufgen/features.go

```diff
@@ -22,86 +22,189 @@ import (
 	"github.com/bufbuild/buf/private/bufpkg/bufconfig"
 	"github.com/bufbuild/buf/private/bufpkg/bufimage"
 	"github.com/bufbuild/buf/private/pkg/app"
+	"google.golang.org/protobuf/reflect/protoreflect"
 	"google.golang.org/protobuf/types/descriptorpb"
 	"google.golang.org/protobuf/types/pluginpb"
 )
 
 // requiredFeatures maps a feature to the set of files in an image that
 // make use of that feature.
-type requiredFeatures map[pluginpb.CodeGeneratorResponse_Feature][]string
+type requiredFeatures struct {
+	flags                  map[pluginpb.CodeGeneratorResponse_Feature][]string
```

Comment:
I don't know what flags actually are - regardless, this variable should be named `featureToXs`, where `X` is what the values are